packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx

```diff
@@ -0,0 +1,542 @@
+  const table = useReactTable({
+    data,
+    columns,
+    state: {
+      rowSelection,
+      columnFilters,
+    },
+    enableRowSelection: isEditable,
+    onRowSelectionChange: setRowSelection,
+    onColumnFiltersChange: setColumnFilters,
+    getCoreRowModel: getCoreRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFacetedRowModel: getFacetedRowModel(),
+    getFacetedUniqueValues: getFacetedUniqueValues(),
+    getFacetedMinMaxValues: getFacetedMinMaxValues(),
+    // debugTable: true,
+    // debugHeaders: true,
+    // debugColumns: false,
+    columnResizeMode: 'onChange',
+  })
+
+  if (isLoading) {
+    return <SkeletonTable numCols={3} numRows={10} />
+  } else if (!isSuccess) {
+    return <></>
+  }
+  const onRemoveFromAR: MouseEventHandler<HTMLButtonElement> = e => {
+    // rowSelection looks like {3: true. 5: true} where the key is the row index.
+    // Create a new ReferenceList based on the entityHeaders in the current table.
+    const updatedData = data.filter(
+      (_value, index) => !(rowSelection[index] === true),
+    )
+    const newRowRefs: ReferenceList = updatedData.map(entityHeader => {
+      return {
+        targetId: entityHeader.id,
+      }
+    })
+    updateRefsInState(newRowRefs)
+  }
+  const selectableTypes = [
+    EntityType.PROJECT,
+    EntityType.FOLDER,
+    EntityType.FILE,
+    EntityType.TABLE,
+    EntityType.LINK,
+    EntityType.ENTITY_VIEW,
+    EntityType.DOCKER_REPO,
+    EntityType.SUBMISSION_VIEW,
+    EntityType.DATASET,
+    EntityType.DATASET_COLLECTION,
+    EntityType.MATERIALIZED_VIEW,
+    EntityType.VIRTUAL_TABLE,
+  ]
+
+  const isSelection = selectionCount > 0
+  const totalRowCount = data.length
+  const filteredRowCount = table.getPrePaginationRowModel().rows.length
+  return (
+    <div>
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          padding: '12px 10px 10px 5px',
+        }}
+      >
+        <div>
+          {totalRowCount > UNMANAGEABLE_SUBJECT_COUNT && (
+            <Typography variant="body1" sx={{ marginBottom: '10px' }}>
+              {totalRowCount} Entities{' '}
+              {filteredRowCount < totalRowCount
+                ? `(${filteredRowCount} visible)`
+                : ''}
+              {isSelection && <span>{` (${selectionCount} selected)`}</span>}
+            </Typography>
+          )}
+        </div>
+        <div>
+          {isEditable && (
+            <Button
+              variant="contained"
+              disabled={!isSelection}
+              onClick={onRemoveFromAR}
+            >
+              {removeSelectedRowsButtonText}
+            </Button>
+          )}
+        </div>
+      </Box>
+      {totalRowCount > 0 && (
+        <Box
+          sx={{
+            overflow: 'auto',
+            maxHeight: '250px',
+            paddingLeft: '2px',
+            th: {
+              backgroundColor: '#eee',
```

Comment:
Looks like you override this below, unless it helps with sticky styling somehow?